ArduCopter/takeoff.cpp

```diff
@@ -111,6 +111,9 @@ void Mode::auto_takeoff_run()
     if (!copter.failsafe.radio && copter.flightmode->use_pilot_yaw()) {
         // get pilot's desired yaw rate
         target_yaw_rate = get_pilot_desired_yaw_rate(channel_yaw->get_control_in());
+        if (!is_zero(target_yaw_rate)) {
+            auto_yaw.set_mode(AUTO_YAW_HOLD);
+        }
```

Comment:
We could add a check for takeoff_complete to reduce the risk of yawing on the ground.